File: src/server/domain/objects/sorters/sort_manager.py
from domain.objects.exceptions.base import BaseSorterException
from domain.objects.sorters.sorter import Comparison, Sorter
from domain.objects.sorters.merge import MergeSorter
from domain.objects.sorters.queue_merge import QueueMergeSorter

class AlgorithmNotFoundException(BaseSorterException):
    errorCode = 404
    def __init__(self, name: str) -> None:
        super().__init__(f"Sorting algorithm not found: {name}.")

def getSortingAlgorithm(
        name: str,
        history: list[Comparison],
        deleted: list[Comparison],
        seed: int
    ) -> Sorter:

    if (name == MergeSorter.SORT_NAME):
        return MergeSorter(history, deleted, seed)
    elif (name == QueueMergeSorter.SORT_NAME):
        return QueueMergeSorter(history, deleted, seed)
    else:
        raise AlgorithmNotFoundException(name)

File: app/server/src/domain/sort-manager.ts
import { AlgorithmTypes } from "@sorter/api";
import { BaseException } from "./exceptions/base.js"
import { Comparison } from "./objects/comparison.js";
import { MergeSorter } from "./sorters/merge.js";
import { QueueMergeSorter } from "./sorters/queue-merge.js";

class AlgorithmNotFoundException extends BaseException {
    constructor(sortType: AlgorithmTypes) {
        super("NOT_FOUND", `Sorting algorithm not found: "${sortType}".`);
    }
}

export function getSortingAlgorithm(name: string, history: Comparison[], deleted: Comparison[], seed: number) {
    const sortType: AlgorithmTypes = name as AlgorithmTypes;
    if (sortType === AlgorithmTypes.QUEUE_MERGE) {
        return new QueueMergeSorter(history, deleted, seed);
    }
    else if (sortType === AlgorithmTypes.MERGE) {
        return new MergeSorter(history, deleted, seed);
    }
    else {
        throw new AlgorithmNotFoundException(sortType);
    }
}
